webserver/static/scripts/datasets/eval-jobs-viewer.js

```diff
@@ -165,12 +208,20 @@ var JobRow = React.createClass({
         id: React.PropTypes.string.isRequired,
         created: React.PropTypes.string.isRequired,
         status: React.PropTypes.string.isRequired,
-        onViewDetails: React.PropTypes.func.isRequired
+        showDelete: React.PropTypes.bool.isRequired,
+        onViewDetails: React.PropTypes.func.isRequired,
+        onDelete: React.PropTypes.func.isRequired
     },
     handleViewDetails: function (event) {
         event.preventDefault();
         this.props.onViewDetails(this.props.index);
     },
+    handleDelete: function (event) {
+        event.preventDefault();
+        if (confirm("Are you sure you want to delete evaluation job " + this.props.id + "?")) {
```

Comment:
yes, that's true, but it feels too messy. 

You could replace the contents of the `<tr controls>` with an inline confirmation